src/library/autodj/autodjfeature.cpp

```diff
@@ -164,98 +161,66 @@ bool AutoDJFeature::dragMoveAccept(QUrl url) {
 }
 
 // Add a crate to the auto-DJ queue.
-void AutoDJFeature::slotAddCrateToAutoDj(int crateId) {
-    m_crateDao.setCrateInAutoDj(crateId, true);
+void AutoDJFeature::slotAddCrateToAutoDj(int iCrateId) {
+    Crate crate;
+    if (m_pTrackCollection->crates().readCrateById(CrateId(iCrateId), &crate) && !crate.isAutoDjSource()) {
+        crate.setAutoDjSource(true);
+        m_pTrackCollection->updateCrate(crate);
+    }
 }
 
 void AutoDJFeature::slotRemoveCrateFromAutoDj() {
-    // Get the crate that was right-clicked on.
     QString crateName = m_lastRightClickedIndex.data().toString();
 
-    // Get the ID of that crate.
-    int crateId = m_crateDao.getCrateIdByName(crateName);
-
-    // Clear its auto-DJ status.
-    m_crateDao.setCrateInAutoDj(crateId, false);
-}
-
-void AutoDJFeature::slotCrateAdded(int crateId) {
-    // If this newly-added crate is in the auto-DJ queue, add it to the list.
-    if (m_crateDao.isCrateInAutoDj(crateId)) {
-        slotCrateAutoDjChanged(crateId, true);
+    Crate crate;
+    if (m_pTrackCollection->crates().readCrateByName(crateName, &crate) && crate.isAutoDjSource()) {
+        crate.setAutoDjSource(false);
+        m_pTrackCollection->updateCrate(crate);
```

Comment:
This code is very similar to the code above.
What is the reason to expose the code here instead of wrapping it to crateDao?